azext_iot/digitaltwins/providers/connection/builders.py

```diff
@@ -0,0 +1,285 @@
+    TRY_ADD_ROLE_LOG_MSG,
+    PRESENT_ADD_ROLE_LOG_MSG,
+    FINISHED_ADD_ROLE_LOG_MSG,
+    SKIP_MSG,
+    FAIL_RBAC_MSG,
+    FAIL_GENERIC_MSG,
+    ABORT_MSG,
+    ADD_INPUT_MSG,
+    CONT_INPUT_MSG
+)
+
+logger = get_logger(__name__)
+
+
+class AdxConnectionValidator(ABC):
+    def __init__(
+        self,
+        adx_cluster_name: str,
+        adx_database_name: str,
+        adx_resource_group: str,
+        adx_subscription: str,
+        dt_instance,
+        eh_namespace: str,
+        eh_entity_path: str,
+        eh_resource_group: str,
+        eh_subscription: str,
+        eh_consumer_group: str = '$Default',
+        yes: bool = False,
+    ):
+        self.cli = EmbeddedCLI()
+        self.yes = yes
+        self.dt = dt_instance
+        if self.dt.identity is None:
+            raise CLIError(DT_IDENTITY_ERROR)
+
+        # Populate adx_cluster_uri, adx_location, adx_resource_id and perform checks
+        self.validate_adx(
+            adx_cluster_name=adx_cluster_name,
+            adx_database_name=adx_database_name,
+            adx_resource_group=adx_resource_group,
+            adx_subscription=adx_subscription,
+        )
+
+        self.eh_consumer_group = eh_consumer_group
+        # Populate eh_endpoint_uri, eh_namespace_resource_id and perform checks
+        self.validate_eventhub(
+            eh_namespace=eh_namespace,
+            eh_entity_path=eh_entity_path,
+            eh_resource_group=eh_resource_group,
+            eh_subscription=eh_subscription,
+            eh_consumer_group=eh_consumer_group,
+        )
+
+    def validate_eventhub(
+        self,
+        eh_namespace: str,
+        eh_entity_path: str,
+        eh_resource_group: str,
+        eh_subscription: str,
+        eh_consumer_group: str
+    ):
+        from azext_iot.digitaltwins.providers.endpoint.builders import EventHubEndpointBuilder
+        eh_endpoint = EventHubEndpointBuilder(
+            endpoint_resource_name=eh_entity_path,
+            endpoint_resource_group=eh_resource_group,
+            endpoint_resource_namespace=eh_namespace,
+            endpoint_resource_policy=None,
+            endpoint_subscription=eh_subscription,
+        )
+        eh_endpoint.error_prefix = ERROR_PREFIX + " find"
+        self.eh_endpoint_uri = eh_endpoint.build_identity_based().endpoint_uri
+
+        eh_consumer_group_op = self.cli.invoke(
+            "eventhubs eventhub consumer-group show -n {} --eventhub-name {} --namespace-name {} -g {}".format(
+                eh_consumer_group,
+                eh_entity_path,
+                eh_namespace,
+                eh_resource_group,
+            ),
+            subscription=eh_subscription,
+        )
+        if not eh_consumer_group_op.success():
+            raise CLIError("{} retrieve Event Hub Consumer Group.".format(ERROR_PREFIX))
+
+        self.eh_namespace_resource_id = (
+            "/subscriptions/{}/resourceGroups/{}/providers/Microsoft.EventHub/namespaces/{}".format(
```

Comment:
Ideally the endpoint builder supports providing the fully qualified Id as it already has all the information.

Reply:
would require code changes in the endpoint builder but the dt endpoint api has no need for the resource id. Only the data history requires the resource ids

Reply:
made a task for the future